landscape/client/monitor/snapmonitor.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from landscape.client.monitor.plugin import DataWatcher
+from landscape.client.snap.http import SnapdHttpException
+from landscape.client.snap.http import SnapHttp
+
+
+class SnapMonitor(DataWatcher):
+
+    run_interval = 1800  # 30 minutes
+    message_type = "snaps"
+    message_key = message_type
+    persist_name = message_type
+    scope = "snaps"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._snap_http = SnapHttp()
+
+    def register(self, registry):
+        self.config = registry.config
+        self.run_interval = self.config.snap_monitor_interval
+
+        super(SnapMonitor, self).register(registry)
+
+    def get_data(self):
+        return get_installed_snaps(self._snap_http)
```

Comment:
The DataWatcher class will cache the get_data value but since it includes timestamps, it wont so probably best to remove them and sort keys, (even though eventually in the message they get stripped out later).

Reply:
Good point. They only get stripped once they are put in the message. I _could_ store the _message_, though. Thinking about it. I will adjust either way.